programs/pluto/kernel_xfrm.c

```diff
@@ -1307,10 +1324,18 @@ static bool netlink_add_sa(const struct kernel_state *sa, bool replace,
 	req.p.reqid = sa->reqid;
 	dbg("%s() adding IPsec SA with reqid %d", __func__, sa->reqid);
 
-	req.p.lft.soft_byte_limit = sa->sa_max_soft_bytes;
-	req.p.lft.hard_byte_limit = sa->sa_ipsec_max_bytes;
-	req.p.lft.hard_packet_limit = sa->sa_ipsec_max_packets;
-	req.p.lft.soft_packet_limit = sa->sa_max_soft_packets;
+	if (!sa->nic_offload.dev || sa->nic_offload.type != OFFLOAD_PACKET)
+	{
+		req.p.lft.soft_byte_limit = sa->sa_max_soft_bytes;
+		req.p.lft.hard_byte_limit = sa->sa_ipsec_max_bytes;
+		req.p.lft.hard_packet_limit = sa->sa_ipsec_max_packets;
+		req.p.lft.soft_packet_limit = sa->sa_max_soft_packets;
+	} else {
+		req.p.lft.soft_byte_limit = XFRM_INF;
+		req.p.lft.hard_byte_limit = XFRM_INF;
+		req.p.lft.hard_packet_limit = XFRM_INF;
+		req.p.lft.soft_packet_limit = XFRM_INF;
+	}
```

Comment:
If this is a limitation, we should have the connection load already fail on these, except that undermines the usefulness of "auto".

I'm also not sure about this messing with FIPS. We set limited to ensure in-time rekeying but if that is not supported we have to "work around" it with timer based rekeys. I guess for AES that does not matter too much,
since the limits more than 2^32 ? But perhaps at least a warning/debug message should be logged.

What happens when

Reply:
sounds good to me, i guess a debug/warning the way to go, till some supporting HW/implementation is available.